core/templates/components/entity-creation-services/topic-creation.service.ts

```diff
@@ -40,16 +47,34 @@ angular.module('oppia').factory('TopicCreationService', [
         if (topicCreationInProgress) {
           return;
         }
-
-        $uibModal.open({
+        var modalInstance = $uibModal.open({
```

Comment:
Should we be declaring this as a separate controller? /cc @marianazangrossi @nithusha21

Reply:
Yeah we should. Done.